File: test_data/csharp/test_main/aas_core_meta.v3rc2/input/snippets/Verification/data_specification_IEC_61360s_for_document_have_appropriate_data_type.cs
/// <summary>
/// Check that the <see cref="Aas.DataSpecificationIec61360.DataType" /> is defined
/// appropriately for all data specifications whose content is given as IEC 61360.
/// </summary>
[CodeAnalysis.SuppressMessage("ReSharper", "InconsistentNaming")]
public static bool DataSpecificationIec61360sForDocumentHaveAppropriateDataType(
    IEnumerable<Aas.EmbeddedDataSpecification> embeddedDataSpecifications
)
{
    foreach (var embeddedDataSpecification in embeddedDataSpecifications)
    {
        var iec61360 = (
            embeddedDataSpecification.DataSpecificationContent
                as DataSpecificationIec61360
        );
        if (iec61360 != null)
        {
            if (
                iec61360.DataType == null
                || !Constants.DataTypeIec61360ForDocument.Contains(
                    iec61360.DataType)
            )
            {
                return false;
            }
        }
    }

    return true;
}

File: dev/test_data/main/java/expected/aas_core_meta.v3/input/snippets/Verification/data_specification_IEC_61360s_for_document_have_appropriate_data_type.java
/**
* Check that the {@link  IDataSpecificationIec61360#getDataType dataType} is defined
* appropriately for all data specifications whose content is given as IEC 61360.
*/
public static boolean dataSpecificationIec61360sForDocumentHaveAppropriateDataType(
  Iterable<? extends IEmbeddedDataSpecification> embeddedDataSpecifications) {
  Objects.requireNonNull(embeddedDataSpecifications);

  for (IEmbeddedDataSpecification embeddedDataSpecification : embeddedDataSpecifications) {
    IDataSpecificationIec61360 iec61360 =
      (IDataSpecificationIec61360) embeddedDataSpecification.getDataSpecificationContent();
    if(iec61360 != null) {
      if (!iec61360.getDataType().isPresent()
        || !Constants.dataTypeIec61360ForDocument.contains(
           iec61360.getDataType().get())) {
        return false;
      }
    }
  }
  return true;
}
